### PR TITLE
Include refocus logging client

### PR DESCRIPTION
**Pull request by guptad2**

### CODE REVIEW

File: cache/models/samples.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ function parseName(name) {
   }
 
   logger.error(`cache/models/samples.parseName|Invalid sample name "${name}"`);
-  console.trace(); // eslint-disable-line no-console
+  logger.verbose();
```
Comment (iamigo) [inline cache/models/samples.js]: `logger.verbose(new Error().stack)`

File: cache/sampleStoreInit.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ const Subject = require('../db').Subject;
 const featureToggles = require('feature-toggles');
 const redisClient = require('./redisCache').client.sampleStore;
 const samsto = require('./sampleStore');
-const log = require('winston');
+const log = require('@salesforce/refocus-logging-client');
```
Comment (iamigo) [inline cache/sampleStoreInit.js]: let's switch this var name to "logger" just for consistency across all the other modules

File: db/migrate.js [hunk 3]
```diff
@@ -13,20 +13,21 @@
  * current migration(s).
  */
 const exec = require('child_process').exec;
+const logger = require('@salesforce/refocus-logging-client');
 const cmd = 'node_modules/.bin/sequelize db:migrate';
 
 module.exports = exec(cmd, (error, stdout, stderr) => {
   // Log the results to the console.
-  console.log('[./db/migrate (stdout)] ' + // eslint-disable-line no-console
+  logger.info('[./db/migrate (stdout)] ' +
```
Comment (iamigo) [inline db/migrate.js]: combine lines, i.e. no carriage return necessary

File: package.json [hunk 4]
```diff
@@ -73,6 +73,7 @@
   },
   "dependencies": {
     "@salesforce/refocus-collector": "^1.2.1",
+    "@salesforce/refocus-logging-client": "^1.0.7",
```
Comment (iamigo) [inline package.json]: run npm update @salesforce/refocus-logging-client

File: realtime/pubSubStats.js [hunk 5]
```diff
@@ -69,7 +70,7 @@ function track(src, evt, obj) {
     nameFromObj = obj.new.name;
     elapsed = now - new Date(obj.new.updatedAt);
   } else {
-    console.trace('Where is updatedAt? ' + JSON.stringify(obj));
+    logger.verbose('Where is updatedAt? ' + JSON.stringify(obj));
```
Comment (iamigo) [inline realtime/pubSubStats.js]: add the stack trace e.g.
`logger.verbose('Where is updatedAt? ' + JSON.stringify(obj) + '\n' + new Error().stack);`

File: realtime/redisPublisher.js [hunk 6]
```diff
@@ -51,17 +51,17 @@ function trackStats(key, obj) {
     obj.new.hasOwnProperty('updatedAt')) {
     elapsed = Date.now() - new Date(obj.new.updatedAt);
   } else {
-    console.trace('Where is updatedAt? ' + JSON.stringify(obj));
+    logger.verbose('Where is updatedAt? ' + JSON.stringify(obj));
```
Comment (iamigo) [inline realtime/redisPublisher.js]: add the stack trace

File: realtime/utils.js [hunk 7]
```diff
@@ -391,7 +393,7 @@ function attachAspectSubject(sample, subjectModel, aspectModel) {
   // check if sample object contains name
   if (!sample.name || sample.name.indexOf('|') < 0) {
     logger.error('sample object does not contain name', JSON.stringify(sample));
-    console.trace('from attachAspectSubject');
+    logger.verbose('from attachAspectSubject');
```
Comment (iamigo) [inline realtime/utils.js]: add stack trace

File: tests/api/v1/lenses/get.js [hunk 8]
```diff
@@ -100,7 +101,7 @@ describe('tests/api/v1/lenses/get.js >', () => {
           }
 
           const obj = res.body;
-          console.log(obj);
+          logger.info(obj);
```
Comment (iamigo) [inline tests/api/v1/lenses/get.js]: should just remove this... test should not be console logging anything

File: tests/api/v1/subjects/post.js [hunk 9]
```diff
@@ -610,7 +611,7 @@ describe(`tests/api/v1/subjects/post.js, POST ${path} >`, () => {
         })
         .end((err /* , res */) => {
           if (err) {
-            console.log(err);
+            logger.info(err);
```
Comment (iamigo) [inline tests/api/v1/subjects/post.js]: no console.log in tests

File: tests/db/model/token/utils.js [hunk 10]
```diff
@@ -63,6 +64,6 @@ module.exports = {
       });
       return tu.db.Token.bulkCreate(toCreate);
     })
-    .catch(console.error);
+    .catch(logger.error);
```
Comment (iamigo) [inline tests/db/model/token/utils.js]: probably don't need this

File: utils/common.js [hunk 11]
```diff
@@ -25,9 +26,9 @@ function logInvalidHmsetValues(key, obj) {
     for (const _key in obj) {
       if (_key === null || obj[_key] === null ||
         (obj[_key] === undefined) || Array.isArray(obj[_key])) {
-        // eslint-disable-next-line no-console
+        // eslint-disable-next-line
```
Comment (iamigo) [inline utils/common.js]: remove this line

File: utils/common.js [hunk 12]
```diff
@@ -25,9 +26,9 @@ function logInvalidHmsetValues(key, obj) {
...
         const stringified = JSON.stringify(obj, (k, v) => v === undefined ? 'undefined' : v);
-        console.trace('Invalid hmset params: key ' + key +
+        logger.verbose('Invalid hmset params: key ' + key +
```
Comment (iamigo) [inline utils/common.js]: add stack trace

File: worker/jobs/executeClockJob.js [hunk 13]
```diff
@@ -31,7 +32,7 @@ module.exports = (job, done) => {
    * "Error: Cannot find module '../../clock/scheduledJobs/undefined'".
    */
   if (!clockJobName) {
-    console.trace('Missing Clock Job Name', job);
+    logger.verbose('Missing Clock Job Name', job);
```
Comment (iamigo) [inline worker/jobs/executeClockJob.js]: add stack trace

File: tests/db/model/token/utils.js (hunk 10 above)
Comment (guptad2) [reply tests/db/model/token/utils.js]: So what to add in the catch? Just an empty function or remove the catch entirely?

Comment (iamigo) [reply tests/db/model/token/utils.js]: remove the catch entirely. the test failed.

Comment (guptad2) [reply tests/db/model/token/utils.js]: Fixed.

**Comment by guptad2** [comment]
@pallavi2209 Checked, it works the logs shown before are still being shown for the local logging part. Also there are some logging related tests which would other wise fail if the logs produced now were different.

File: migrations/20190603225918-no-paranoid.js [hunk 14]
```diff
@@ -14,6 +14,7 @@
  * *after* the "destroySoftDeleted" step has completed!
  */
 'use strict';
+const logger = require('@salesforce/refocus-logging-client');
```
Comment (jgraff2) [inline migrations/20190603225918-no-paranoid.js]: Not needed for migrations.

File: tests/clock/utils/utils.js [hunk 15]
```diff
@@ -11,6 +11,7 @@
  */
 
 'use strict'; // eslint-disable-line strict
+const logger = require('@salesforce/refocus-logging-client');
```
Comment (jgraff2) [inline tests/clock/utils/utils.js]: Not needed in test files.

File: migrations/20190603225918-no-paranoid.js (hunk 14 above)
Comment (guptad2) [reply migrations/20190603225918-no-paranoid.js]: fixed

File: tests/clock/utils/utils.js (hunk 15 above)
Comment (guptad2) [reply tests/clock/utils/utils.js]: fixed

**Pull request merged**